Windows build broken at cli - System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\build\BranchInfo.props'.

Tested latest source-build and it has failed during CLI repo build. Seems that some directories are hard coded and do not respect root of repo to create correct relative layout.

Error details:

```
Target "Update" in project "E:\src\ms\dotnet\source-build\targets\repository.proj" (target "Build" depends on it):
Task "Exec"
  Environment Variables passed to tool:
    DOTNET_RUNTIME_ID=win10-x64
    DOTNET_TOOL_DIR=E:\src\ms\dotnet\source-build\Tools/dotnetcli/
    BUILD_TOOLS_TOOL_DIR=E:\src\ms\dotnet\source-build\Tools/
    BUILDTOOLS_SKIP_CROSSGEN=1
    DotNetBuildFromSource=true
    DotNetRestorePackagesPath=E:\src\ms\dotnet\source-build\packages/
    DotNetCoreSdkDir=E:\src\ms\dotnet\source-build\Tools/dotnetcli/
    DotNetBuildToolsDir=E:\src\ms\dotnet\source-build\Tools/
    GitInfoCommitCount=6908
    GitInfoCommitHash=38031f6e5c3e21fa39d82aca300f5ed769be10a0
  (set DOTNET_VERSION_URL=file://E:\src\ms\dotnet\source-build\bin/obj/x64/Release/build-info/) & powershell.exe E:\src\ms\dotnet\source-build\src/cli/build_projects/update-dependencies/update-dependencies.ps1 -Update
  
  PowerShell version: 5.1.15063.632
  Free physical memory: 11997 MB out of 16300 MB

  Restoring E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\update-dependencies.csproj...
    Restoring packages for E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\update-dependencies.csproj...
    Installing Microsoft.DotNet.VersionTools 1.0.27-prerelease-01723-01.
    Generating MSBuild file E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\obj\update-dependencies.csproj.nuget.g.props.
    Generating MSBuild file E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\obj\update-dependencies.csproj.nuget.g.targets.
    Restore completed in 4,07 sec for E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\update-dependencies.csproj.
  Invoking App E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\update-dependencies.csproj...

  Unhandled Exception: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\build\BranchInfo.props'.
     at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
     at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
     at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
     at System.Xml.XmlDownloadManager.GetStream(Uri uri, ICredentials credentials, IWebProxy proxy, RequestCachePolicy cachePolicy)
     at System.Xml.XmlUrlResolver.GetEntity(Uri absoluteUri, String role, Type ofObjectToReturn)
     at System.Xml.XmlTextReaderImpl.FinishInitUriString()
     at System.Xml.XmlTextReaderImpl..ctor(String uriStr, XmlReaderSettings settings, XmlParserContext context, XmlResolver uriResolver)
     at System.Xml.XmlReaderSettings.CreateReader(String inputUri, XmlParserContext inputContext)
     at System.Xml.XmlReader.Create(String inputUri, XmlReaderSettings settings)
     at System.Xml.Linq.XDocument.Load(String uri, LoadOptions options)
     at System.Xml.Linq.XDocument.Load(String uri)
     at Microsoft.DotNet.Scripts.Config.GetRepoMSBuildPropValue(String propsFileName, String propertyName) in E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\Config.cs:line 94
     at Microsoft.DotNet.Scripts.Config.GetDefaultUpstreamBranch() in E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\Config.cs:line 80
     at Microsoft.DotNet.Scripts.Config.<>c.<.ctor>b__12_7() in E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\Config.cs:line 44
     at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
     at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
     at System.Lazy`1.CreateValue()
     at Microsoft.DotNet.Scripts.Config.get_GitHubUpstreamBranch() in E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\Config.cs:line 59
     at Microsoft.DotNet.Scripts.Config.GetDefaultCoreSetupVersionFragment() in E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\Config.cs:line 86
     at Microsoft.DotNet.Scripts.Config.<>c.<.ctor>b__12_4() in E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\Config.cs:line 41
     at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
     at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
     at System.Lazy`1.CreateValue()
     at Microsoft.DotNet.Scripts.Config.get_CoreSetupVersionFragment() in E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\Config.cs:line 56
     at Microsoft.DotNet.Scripts.Program.Main(String[] args) in E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\Program.cs:line 28
  Build failed
  At E:\src\ms\dotnet\source-build\src\cli\build_projects\update-dependencies\update-dependencies.ps1:51 char:27
  + if($LASTEXITCODE -ne 0) { throw "Build failed" }
  +                           ~~~~~~~~~~~~~~~~~~~~
      + CategoryInfo          : OperationStopped: (Build failed:String) [], RuntimeException
      + FullyQualifiedErrorId : Build failed

E:\src\ms\dotnet\source-build\targets\repository.proj(100,5): error MSB3073: The command "(set DOTNET_VERSION_URL=file://E:\src\ms\dotnet\source-build\bin/obj/x64/Release/build-info/) & powershell.exe E:\src\ms\dotnet\source-build\src/cli/build_projects/update-dependencies/update-dependencies.ps1 -Update" exited with code 1.
Done executing task "Exec" -- FAILED.
Done building target "Update" in project "repository.proj" -- FAILED.
Done Building Project "E:\src\ms\dotnet\source-build\targets\repository.proj" (BuildRepositoryAndDependencies target(s)) -- FAILED.
Done executing task "MSBuild" -- FAILED.
Done building target "Build" in project "build.proj" -- FAILED.
Done Building Project "E:\src\ms\dotnet\source-build\build.proj" (default targets) -- FAILED.

Build FAILED.

```

System details:

Windows 10 Pro 1703 x64 pl-PL with language packs defaults to en-US
Visual Studio 2017 v15.3.5

--- a/targets/cli.props
+++ b/targets/cli.props
@@ -9,7 +9,7 @@
     <TarBallOutput>$(PackagesOutput)</TarBallOutput>
     <GitCommitCount>6908</GitCommitCount>
     <GitCommitHash>38031f6e5c3e21fa39d82aca300f5ed769be10a0</GitCommitHash>
-    <UpdateCommand>(set DOTNET_VERSION_URL=file:%2F%2F$(LocalBuildInfoRoot)) &amp; powershell.exe $(ProjectDirectory)build_projects/update-dependencies/update-dependencies.ps1 -Update</UpdateCommand>
+    <UpdateCommand>(set DOTNET_VERSION_URL=file:%2F%2F$(LocalBuildInfoRoot)) &amp; powershell.exe -NoProfile $(ProjectDirectory)build_projects/update-dependencies/update-dependencies.ps1 -Update</UpdateCommand>
     <UpdateCommand Condition="'$(OS)' != 'Windows_NT'">DOTNET_VERSION_URL=file:%2F%2F$(LocalBuildInfoRoot) $(ProjectDirectory)build_projects/update-dependencies/update-dependencies$(ShellExtension) --Update</UpdateCommand>
     <RepoApiImplemented>false</RepoApiImplemented>
   </PropertyGroup>

